libethereum/State.cpp

```diff
@@ -589,8 +594,16 @@ void State::rollback(size_t _savepoint)
             account.resetCode();
             break;
         case Change::Touch:
-            account.untouch();
-            m_unchangedCacheEntries.emplace_back(change.address);
+            // Empty RIPEMD contract needs to be deleted even in case of exception.
+            // This doesn't affect main net, but is needed for some consensus tests.
+            // https://github.com/ethereum/go-ethereum/pull/3341/files#diff-2433aa143ee4772026454b8abd76b9dd
+            // https://github.com/ethereum/EIPs/issues/716
+            // https://github.com/ethereum/aleth/pull/5652
+            if (change.address != c_RipemdPrecompiledAddress)
```

Comment:
Uff, this check is actually placed in some hot part of the code...

Reply:
Maybe we could try to introduce some new kind of touch, that is un-revertable and called only for precompiles...

Reply:
@chfast Ok, here's the alternative solution https://github.com/ethereum/aleth/pull/5663
No hacks in `rollback()` at the expense of additional flag in `Account` class.

I think I like it better, at least it allows to have one central piece of weirdness only in `Executive`.

Reply:
Actually now I have a third idea, to keep the list of unrevertably touched accounts in `State`, instead of putting the flag in each `Account`.
Third PR is coming...

Reply:
https://github.com/ethereum/aleth/pull/5664